{
  "configurations": {
    "CVE_data_version": "4.0",
    "nodes": []
  },
  "cve": {
    "CVE_data_meta": {
      "ASSIGNER": "cve@kernel.org",
      "ID": "CVE-2024-45027"
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "In the Linux kernel, the following vulnerability has been resolved:\n\nusb: xhci: Check for xhci-\u003einterrupters being allocated in xhci_mem_clearup()\n\nIf xhci_mem_init() fails, it calls into xhci_mem_cleanup() to mop\nup the damage. If it fails early enough, before xhci-\u003einterrupters\nis allocated but after xhci-\u003emax_interrupters has been set, which\nhappens in most (all?) cases, things get uglier, as xhci_mem_cleanup()\nunconditionally derefences xhci-\u003einterrupters. With prejudice.\n\nGate the interrupt freeing loop with a check on xhci-\u003einterrupters\nbeing non-NULL.\n\nFound while debugging a DMA allocation issue that led the XHCI driver\non this exact path."
        }
      ]
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": []
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "name": "https://git.kernel.org/stable/c/770cacc75b0091ece17349195d72133912c1ca7c",
          "refsource": "",
          "tags": [],
          "url": "https://git.kernel.org/stable/c/770cacc75b0091ece17349195d72133912c1ca7c"
        },
        {
          "name": "https://git.kernel.org/stable/c/dcdb52d948f3a17ccd3fce757d9bd981d7c32039",
          "refsource": "",
          "tags": [],
          "url": "https://git.kernel.org/stable/c/dcdb52d948f3a17ccd3fce757d9bd981d7c32039"
        }
      ]
    }
  },
  "impact": {},
  "lastModifiedDate": "2024-09-11T16:26Z",
  "publishedDate": "2024-09-11T16:15Z"
}